core/src/main/scala/chisel3/Aggregate.scala

```diff
@@ -1088,12 +1088,50 @@ abstract class Bundle(implicit compileOptions: CompileOptions) extends Record {
     * }}}
     */
   final lazy val elements: SeqMap[String, Data] = {
+    val hardwareFields = _elementsImpl.flatMap {
+      case (name, data: Data) =>
+        if (data.isSynthesizable) {
+          Some(s"$name: $data")
+        } else {
+          None
+        }
+      case (name, data: Some[Data]) =>
+        if (data.get.isSynthesizable) {
+          Some(s"$name: $data")
+        } else {
+          None
+        }
+      case _ => None
+    }
+    if (hardwareFields.nonEmpty) {
+      throw ExpectedChiselTypeException(s"Bundle: $this contains hardware fields: " + hardwareFields.mkString(","))
+    }
+    VectorMap(
+      _elementsImpl
+        .toSeq
+        .flatMap {
+          case (name, data: Data) =>
+              Some(name -> data)
+          case (name, data: Some[Data]) =>
+              Some(name -> data.get)
```

Comment:
```suggestion
          case (name, opt: Some(data: Data)) =>
              Some(name -> data)
```
Due to type erasure, you can't match on the type parameter of the `Some`, but you can unbox it and just look inside.

Reply:
Will be in next commit